packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts

```diff
@@ -0,0 +1,41 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, IHandlerParameters, TextUtils } from "@zowe/imperative";
+import { IZosFilesResponse, ZosFilesMessages } from "../../../api";
+import { List } from "../../../api/methods/list";
+import { ZosFilesBaseHandler } from "../../ZosFilesBase.handler";
+
+/**
+ * Handler to list a data sets
+ * @export
+ */
+export default class ZfsHandler extends ZosFilesBaseHandler {
+    public async processWithSession(commandParameters: IHandlerParameters, session: AbstractSession): Promise<IZosFilesResponse> {
+        const response = await List.zfs(session, {
+            path: commandParameters.arguments.path,
+            fsname: commandParameters.arguments.fsname,
+            maxLength: commandParameters.arguments.maxLength
+        });
+
+        if (commandParameters.arguments.attributes && response.apiResponse.items.length > 0) {
```

Comment:
Few things here. It doesn't look like `attributes` is an option on your command? For now, I wouldn't worry about adding the attributes flag. Instead, you can use the items array returned and formulate a table. You can use the `response.format.output` interface:

```
        commandParameters.response.format.output({
            fields: ["name", "mountpoint"],
            output: response.apiResponse.items,
            format: "table"
        });
```

Basically, you specify your response array on the `output` property and then specify the properties you wish to display by default via `fields`. See [ICommandOutputFormat.ts](https://github.com/zowe/imperative/blob/master/packages/cmd/src/doc/response/response/ICommandOutputFormat.ts) for more details.

You can then allow the user control over the output format (and fields), by enabling `outputFormatOptions: true` on your `ICommandDefinition` for your command.

Also, make sure you place the raw z/OSMF API response payload in the data portion of the response via: `commandParameters.response.data.setObj()`. This way it will be available when the `--response-format-json` flag is used.